code/controllers/subsystem/job.dm

```diff
@@ -321,73 +323,77 @@ SUBSYSTEM_DEF(job)
 	return
 
 
-/**
- * Will try to select a head, ignoring ALL non-head preferences for every level until.
- *
- * Basically tries to ensure there is at least one head in every shift if anyone has that job preference enabled at all.
+/*
+ * Forces a random Head of Staff role to be assigned to a random eligible player.
+ * Returns TRUE if a player was selected and assigned the role. FALSE otherwise.
  */
-/datum/controller/subsystem/job/proc/FillHeadPosition()
+/datum/controller/subsystem/job/proc/force_one_head_assignment()
 	var/datum/job_department/command_department = get_department_type(/datum/job_department/command)
 	if(!command_department)
 		return FALSE
 	for(var/level in level_order)
 		for(var/datum/job/job as anything in command_department.department_jobs)
 			if((job.current_positions >= job.total_positions) && job.total_positions != -1)
 				continue
-			var/list/candidates = FindOccupationCandidates(job, level)
+			var/list/candidates = find_occupation_candidates(job, level)
 			if(!candidates.len)
 				continue
 			var/mob/dead/new_player/candidate = pick(candidates)
-			// Eligibility checks done as part of FindOccupationCandidates.
-			if(AssignRole(candidate, job, do_eligibility_checks = FALSE))
+			// Eligibility checks done as part of find_occupation_candidates.
+			if(assign_role(candidate, job, do_eligibility_checks = FALSE))
 				return TRUE
 	return FALSE
 
 
 /**
  * Attempts to fill out all possible head positions for players with that job at a a given job priority level.
+ * Returns the number of Head positions assigned.
  *
  * Arguments:
- * * level - One of the JP_LOW, JP_MEDIUM or JP_HIGH defines. Attempts to find candidates with head jobs at this priority only.
+ * * level - One of the JP_LOW, JP_MEDIUM, JP_HIGH or JP_ANY defines. Attempts to find candidates with head jobs at that priority only.
  */
-/datum/controller/subsystem/job/proc/CheckHeadPositions(level)
+/datum/controller/subsystem/job/proc/fill_all_head_positions_at_priority(level)
+	. = 0
 	var/datum/job_department/command_department = get_department_type(/datum/job_department/command)
+
 	if(!command_department)
 		return
```

Comment:
Might be nice to have some `return .` here and at the end just to remind future people touching this proc that this proc does have an important return value

Reply:
Can do, explicit code is better than documentation.